airbyte-integrations/connectors/source-myhours/source_myhours/source.py

```diff
@@ -0,0 +1,172 @@
+    ) -> str:
+        return "Reports/activity"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        previous_query = parse_qs(urlparse(response.request.url).query)
+        previous_end = pendulum.parse(previous_query["DateTo"][0])
+
+        new_from = previous_end.add(days=1)
+        new_to = new_from.add(days=self.batch_size - 1)
+
+        if new_from > pendulum.now():
+            return None
+
+        return {
+            "DateFrom": new_from.to_date_string(),
+            "DateTo": new_to.to_date_string(),
+        }
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+
+        if next_page_token is None:
+            return {"DateFrom": self.start_date.to_date_string(), "DateTo": self.start_date.add(days=6).to_date_string()}
+        return next_page_token
+
+
+class Users(MyhoursStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Users/getAll"
+
+
+# Source
+class SourceMyhours(AbstractSource):
+    def get_access_token(self, config) -> Tuple[str, any]:
+        email = config["email"]
+        password = config["password"]
+        url = f"{url_base}/tokens/login"
+
+        try:
+            payload = json.dumps({"grantType": "password", "email": email, "password": password, "clientId": "api"})
+            response = requests.post(url, headers=request_headers, data=payload)
+            response.raise_for_status()
+            json_response = response.json()
+            return json_response.get("accessToken", None), None if json_response is not None else None, None
+        except requests.exceptions.RequestException as e:
+            return None, e
+
+    def check_connection(self, logger: AirbyteLogger, config) -> Tuple[bool, any]:
+        access_token = self.get_access_token(config)
+        url = f"{url_base}/Clients"
+        token_value = access_token[0]
+        token_exception = access_token[1]
+
+        if token_exception:
+            return False, token_exception
+
+        if token_value:
+            headers = TokenAuthenticator(token=token_value).get_auth_header()
+            headers.update(request_headers)
+            try:
+                response = requests.get(url, headers=headers)
+                response.raise_for_status()
+                return True, None
+            except requests.exceptions.RequestException as e:
+                return False, e
+        return False, "Token not found"
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        access_token = self.get_access_token(config)
```

Comment:
I does not look like this token will be refreshed. I'd suggest you to implement a child authenticator from our `Oauth2Authenticator`. Otherwise syncs won't run for more than the token lifetime (1800s = 30mn from My Hours API doc).

Reply:
I did not expect we needed refresh functionality since my api token is valid for 5 days. Since yours is 30 mins, I will implement refresh logic by looking at the implementation in other connector projects.